BitwardenShared/UI/Vault/Vault/VaultList/VaultListProcessor.swift

```diff
@@ -234,6 +258,7 @@ extension VaultListProcessor {
                 do {
                     _ = try await services.notificationService
                         .requestAuthorization(options: [.alert, .sound, .badge])
+                    await self.registerForNotifications()
```

Comment:
🤔 Shouldn't we check the result from `requestAuthorization` and only if it's `true` call `await self.registerForNotifications()`? [Docs](https://developer.apple.com/documentation/usernotifications/unusernotificationcenter/requestauthorization(options:completionhandler:))

Reply:
Good call; I missed that

Reply:
You can register for notifications without yet having authorization (or if the user denied authorization). The benefit of that is if the user enables notifications in settings, they will start receiving notifications immediately without having to reopen the app.